tutorials/graph-theory/prims-algorithm.md

```diff
@@ -9,4 +9,130 @@ keywords:
+The Prims algorithm is a [Greedy Algorithm](../basic-topics/greedy.md)  used to find a minimum spanning tree (MST) for a connected weighted undirected graph. In simpler terms, it finds a subset of the edges that forms a tree that includes *every* vertex, where the total **weight** of all the **edges** in the tree is **minimized**.
+
+### How does it work? 
+- It starts with a single vertex (that can be randomly chosen) and gradually expands the tree by adding the smallest edge that connects a vertex present in the tree to a vertex not yet in the tree.
+- It continues until every vertex is included in the tree.
+- The algorithm uses a priority queue or similar structure to efficiently track the smallest edge that can be added to the tree at each step.
+  
+**Note:** Since the starting vertex is chosen at random, it is possible to have different edges included in the MST for the same graph, but the total edge weight of the MST will still have the same minimum value.
+
+### Complexity and Use-Cases
+When implemented using a priority queue (usually a binary heap), the time complexity of Prim’s algorithm is $O(E + V log V)$, where E is the number of edges and V is the number of vertices. This makes Prim's algorithm highly efficient for **dense** graphs, where the number of edges is close to the maximum possible. For sparse graphs (a graph with a relatively small number of edges compared to the maximum number of possible edges), other algorithms like [Kruskal’s](kruskals-algorithm.md) may perform better.
+
+The space complexity of the Prims Algorithm is $O(V + E)$, where V is the number of vertices and E is the number of edges.
+
+Prim's algorithm is widely used in network design, where the goal is to minimize the cost of building a network while connecting all nodes. Examples include laying cables, designing road networks, or creating efficient communication paths. It is also used as a building block in other algorithms for solving more complex problems.
+
+![Prims Algorithm Image](https://miro.medium.com/max/700/1*7kpPIPcmXr38Juh0umM6fA.jpeg)
+
+## Example [1135 - Connecting Cities With Minimum Cost](https://leetcode.com/problems/connecting-cities-with-minimum-cost/description/?envType=problem-list-v2&envId=minimum-spanning-tree)
+**Note:** The above link requires the Leetcode Subscription. To view the problem, use [this](https://leetcode.ca/all/1135.html) alternate link.
+
+### Instructions: 
+- There are N cities numbered from 1 to N.
+- You are given connections, where each connections[i] = [city1, city2, cost] represents the cost to connect city1 and city2 together. The connections are all bidirectional.
+- Return the minimum cost so that for every pair of cities, there exists a path of connections (possibly of length 1) that connects those two cities together.  The cost is the sum of the connection costs used. If the task is impossible, return -1.
+
+Example: 
+**Input:** N = 3, connections = [[1,2,5],[1,3,6],[2,3,1]]
+**Output:** 6
```

Comment:
1. please format with LaTex syntax. 
2. please check in local dev.  like `Example: Input: N = 3, connections = [[1,2,5],[1,3,6],[2,3,1]] Output: 6` renders in one line.